### PR TITLE
Made scripts more robust

### PR DESCRIPTION
**Pull request by heartsucker**
@joelpurra asked me to take a peek at this and see if I could add some stability to the scripts here. Most of this stuff is pedantic, and there are no real changes to functionality as I don't have a rpi to test any iterations on.

The one big change is not accepting command line arguments to the password hashing script. It's considered bad practice as it allows other processes to see the plaintext password.

### CODE REVIEW

**Comment by ln5** [comment]
Thanks!

I'm cherrypicking most of the robustness fixes, commenting the rest here.

I'm not merging the password change. I appreciate the idea of not exposing passwords in the environment but I don't think it's worth the added dependency on perl.

Happy to hear more voices and a separate pull request for this feature if we should go forward.

File: scripts/on-rpi-boot.sh [hunk 1]
```diff
@@ -21,33 +26,34 @@ Pin-Priority: -10
 EOF
 fi
 
-if [ "$(grep -c "http://archive.raspbian.org/raspbian wheezy" /etc/apt/sources.list)" = 0 ]
+if ! egrep -q '^\s*http://archive\.raspbian\.org/raspbian\s*wheezy' /etc/apt/sources.list
 then
-  echo "deb http://archive.raspbian.org/raspbian wheezy main contrib non-free rpi" >> /etc/apt/sources.list
+  echo 'deb http://archive\.raspbian\.org/raspbian wheezy main contrib non-free rpi' >> /etc/apt/sources.list
 fi
```
Comment (ln5) [inline scripts/on-rpi-boot.sh]: Why would we want backslashes in sources.list?

**Comment by ln5** [comment]
Oh, and when I say "cherrypicking" I mean "cherrypicking to branch improve-robustness". These changes need testing. The 'set -e' change to most scripts for example might make us stop somewhere when we're today saved by the fact that we're not checking exit codes properly. We have some 10-15 of these in other peoples homes and can't really afford bricking them.

Testing of branch 'improve-robustness' highly appreciated!

File: scripts/on-rpi-boot.sh (hunk 1 above)
Comment (heartsucker) [inline scripts/on-rpi-boot.sh:32]: That one is definitely a mistake. I meant to only escape the periods in the regex.

**Comment by heartsucker** [comment]
Yeah, I know I couldn't test (no rpi), so I tried to do best guesses on whether or not the `set -e` was safe.

**Comment by heartsucker** [comment]
Closing this as no longer relevant.

**Pull request closed without merging**